src/compiler/checker.ts

```diff
@@ -5058,9 +5058,21 @@ namespace ts {
                 return factory.createKeywordTypeNode(SyntaxKind.AnyKeyword);
             }
 
+            function shouldUsePlaceholderForProperty(propertySymbol: Symbol, context: NodeBuilderContext) {
+                // Use placeholders for reverse mapped types we've either already descended into, or which
+                // are nested reverse mappings within a mapping over a non-anonymous type. The later is a restriction mostly just to
+                // reduce the blowup in printback size from doing, eg, a deep reverse mapping over `Window`.
+                // Since anonymous types usually come from expressions, this allows us to preserve the output
+                // for deep mappings which likely come from expressions, while truncating those parts which
+                // come from mappings over library functions.
+                return !!(getCheckFlags(propertySymbol) & CheckFlags.ReverseMapped) && (
+                    contains(context.reverseMappedStack, propertySymbol as ReverseMappedSymbol)
+                    || (context.reverseMappedStack?.[0] && !(getObjectFlags(last(context.reverseMappedStack).propertyType) & ObjectFlags.Anonymous)));

```

Comment:
tiny style suggestion: put the `&&` on the next line to highlight that it's the parent of the `||`.
```suggestion
                return !!(getCheckFlags(propertySymbol) & CheckFlags.ReverseMapped) 
                    && (
                     contains(context.reverseMappedStack, propertySymbol as ReverseMappedSymbol)
                     || (context.reverseMappedStack?.[0] && !(getObjectFlags(last(context.reverseMappedStack).propertyType) & ObjectFlags.Anonymous)));

```

Reply:
Also, I don't understand what the last condition means and how it maps back to "nested reverse mapping". Is it any reverse mapping at all? Does `context.reverseMappedStack?.[0]` being defined mean that we're in a nested one?

Reply:
Yep. The stack having members means we're within a reverse mapping (essentially the same as the old context flag) - the last member of the stack is just also looked at now, to see if it's a type we should printback anyway.